R/jobs.R

```diff
@@ -0,0 +1,150 @@
+#' }
+#'
+#' @export
+jobAdd <- function(name, status = "", progressUnits = 0L, actions = NULL, running = FALSE, 
+                   autoRemove = TRUE, show = TRUE) {
+    callFun("addJob", name          = name,
+                      status        = status,
+                      progressUnits = progressUnits,
+                      actions       = actions,
+                      running       = running,
+                      autoRemove    = autoRemove,
+                      show          = show)
+}
+
+#' Remove a Job
+#'
+#' Remove a job from RStudio's Jobs pane.
+#' 
+#' @param job The ID of the job to remove
+#'
+#' @export
+jobRemove <- function(job) {
+    callFun("removeJob", job = job)
+}
+
+#' Set Job Progress
+#'
+#' Updates the progress for a job.
+#' 
+#' @param job The ID of the job to set progress for.
+#' @param units The integer number of total units of work completed so far.
+#'
+#' @export
+jobSetProgress <- function(job, units) {
+    callFun("setJobProgress", job = job, units = units)
+}
+
+#' Add Job Progress
+#'
+#' Adds incremental progress units to a job.
+#' 
+#' @param job The ID of the job to update progress for.
+#' @param units The integer number of new progress units completed.
+#'
+#' @export
+jobAddProgress <- function(job, units) {
+    callFun("addJobProgress", job = job, units = units)
+}
+
+#' Set Job Status
+#'
+#' Update a job's informational status text.
+#'
+#' @param job The ID of the job to update.
+#' @param status Text describing job's new status.
+#'
+#' @export
+jobSetStatus <- function(job, status) {
+    callFun("setJobStatus", job = job, status = status)
+}
+
+
+#' Set Job State
+#'
+#' Changes the state of a job.
+#' 
+#' @param job The ID of the job on which to change state.
+#' @param state The new job state.
+#'
+#' @section States:
+#' The following states are supported:
+#' \describe{
+#'   \item{idle}{The job is waiting to run.}
+#'   \item{running}{The job is actively running.}
+#'   \item{succeeded}{The job has finished successfully.}
+#'   \item{cancelled}{The job was cancelled.}
+#'   \item{failed}{The job finished but did not succeed.}
+#' }
+#'
+#' @export
+jobSetState <- function(job, state = c("idle", "running", "succeeded", "cancelled", "failed")) {
+    callFun("setJobState", job = job, state = state)
+}
+
+
+#' Add Job Output
+#'
+#' Adds text output to a job.
+#' 
+#' @param job The ID of the job that has emitted text.
+#' @param output The text output emitted by the job.
+#' @param error Whether the output represents an error.
+#'
+#' @export
+jobAddOutput <- function(job, output, error = FALSE) {
+    callFun("addJobOutput", job = job, output = output, error = error)
+}
+
+
+#' Run R Script As Job
+#'
+#' Starts an R script as a background job.
+#' 
+#' @param path The path to the R script to be run
+#' @param encoding The text encoding of the script, if known.
+#' @param workingDir The working directory in which to run the job.
```

Comment:
What working directory is used by default (when the parameter is `NULL`)?

Reply:
A good question! It's the parent dir of the script. I'll update the documentation so this is clear.